src/game/objects/har.c

```diff
@@ -2229,15 +2229,18 @@ int har_act(object *obj, int act_type) {
         prefix = 'P';
     }
 
-    if(prefix != 1 && h->endurance < 0) { // Mash to recover from stun faster!
-        h->endurance += 512;
-    }
-
     uint32_t input_staleness = obj->gs->tick - h->input_change_tick;
     if(input_changed) {
         h->input_change_tick = obj->gs->tick;
     }
 
+    if(h->endurance < 0) {
+        if(prefix != 1) { // Mash to recover from stun faster!
```

Comment:
```suggestion
        if(prefix == 'K' || prefix == 'P') { // Mash to recover from stun faster!
```
If I understand the significance of 'prefix' correctly, this is equivalent.
Unless prefix is sometimes equal to `\0` here?

Reply:
I agree locality of logic makes sense here